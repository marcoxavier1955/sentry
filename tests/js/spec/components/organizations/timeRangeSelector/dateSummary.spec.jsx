import React from 'react';

import {mount} from 'enzyme';
import DateSummary from 'app/components/organizations/timeRangeSelector/dateSummary';

// 2017-10-14T02:38:00.000Z
// 2017-10-17T02:38:00.000Z
const start = new Date(1507948680000);
const end = new Date(1508207880000); // National Pasta Day

describe('DateSummary', function() {
  let wrapper;
  let routerContext = TestStubs.routerContext();

  const createWrapper = (props = {}) =>
    mount(<DateSummary useUtc start={start} end={end} {...props} />, routerContext);

  it('renders', async function() {
    wrapper = createWrapper();
    expect(wrapper).toMatchSnapshot();
  });

  it('does not show times when it is midnight for both dates', function() {
    wrapper = createWrapper({
      start: new Date('2017-10-14T00:00:00.000Z'),
      end: new Date('2017-10-17T00:00:00.000Z'),
    });

    expect(wrapper.find('Time')).toHaveLength(0);
  });

  it('does not show end datetime when both dates are midnight and difference is 24 hours', function() {
    wrapper = createWrapper({
      start: new Date('2017-10-14T00:00:00.000Z'),
      end: new Date('2017-10-15T00:00:00.000Z'),
    });

    expect(wrapper.find('Time')).toHaveLength(0);
    expect(wrapper.find('Date')).toHaveLength(1);
  });
});
