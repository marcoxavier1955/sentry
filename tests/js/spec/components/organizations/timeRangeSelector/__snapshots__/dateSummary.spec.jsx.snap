// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateSummary renders 1`] = `
<DateSummary
  end={2017-10-17T02:38:00.000Z}
  start={2017-10-14T02:38:00.000Z}
  useUtc={true}
>
  <Flex
    align="center"
  >
    <Base
      align="center"
      className="css-5ipae5"
    >
      <div
        className="css-5ipae5"
        is={null}
      >
        <DateGroup>
          <div
            className="css-9gjle2-DateGroup eyxmd2r0"
          >
            <Date
              hasTime={true}
            >
              <span
                className="css-621ves-Date eyxmd2r1"
              >
                Oct 14, 2017
              </span>
            </Date>
            <Time>
              <span
                className="css-101687m-Time eyxmd2r2"
              >
                02:38
              </span>
            </Time>
          </div>
        </DateGroup>
        <DateRangeDivider>
          <span
            className="css-1kge5d2-DateRangeDivider eyxmd2r3"
          >
            to
          </span>
        </DateRangeDivider>
        <DateGroup>
          <div
            className="css-9gjle2-DateGroup eyxmd2r0"
          >
            <Date
              hasTime={true}
            >
              <span
                className="css-621ves-Date eyxmd2r1"
              >
                Oct 17, 2017
              </span>
            </Date>
            <Time>
              <span
                className="css-101687m-Time eyxmd2r2"
              >
                02:38
              </span>
            </Time>
          </div>
        </DateGroup>
      </div>
    </Base>
  </Flex>
</DateSummary>
`;
